### PR TITLE
Create vote transaction asset for dpos module - Partially Closes #5608

### LINKED ISSUE #5608: Create DPoS module
### Description

* [x]  Create RegisterAsset which extends BaseAsset
    *   Name should be “register”
    *   Type should be 0
    *   Move asset schema from delegate transaction
    *   Move validateAsset functionality and tests from Delegate transaction
    *   Move applyAsset functionality and tests from Delegate transaction
* [x]    Create VoteAsset which extends BaseAsset
    *   Name should be “vote”
    *   Type should be 1
    *   Move asset schema from vote transaction
    *   Move validateAsset functionality and tests from Vote transaction
    *   Move applyAsset functionality and tests from Vote transaction
    *   Update change of balance to use reducer “token:debit” to lock the balance from the sender
*  [x]   Create UnlockAsset which extends BaseAsset
    *   Name should be “unlock”
    *   Type should be 2
    *   Move asset schema from unlock transaction
    *   Move validateAsset functionality and tests from unlock transaction
    *   Move applyAsset functionality and tests from unlock transaction
    *   Update change of balance to use reducer “token:credit” to return the balance to the sender
*  [x]   Create PoMAsset which extends BaseAsset
    *   Name should be “pom”
    *   Type should be 3
    *   Move asset schema from pom transaction
    *   Move validateAsset functionality and tests from pom transaction
    *   Move applyAsset functionality and tests from pom transaction
*  [x]   Create DPoSModule which extends BaseModule
    *   Move accountSchema “asset” from framework to accountSchema
    *   Name should be “dpos”
    *   Type should be 5
    *   Add validation of genesis config in the constructor
        *   Check activeDelegates, standbyDelegates and delegateListRoundOffset exist and proper type
    *   Move dpos.onBlockFinalized functionality to afterBlockApply step
        *   forgerList will not be managed in DPoS, so not required to be moved
        *   Use consensus.getFinalizedHeight to get the finalized height for cleanup
    *   Move genesis block validation related to DPoS from lisk-genesis to afterGenesisBlockApply
    *   Move updateProductivity logic from dpos/delegate_info to afterBlockApply step
    *   Move applyBootstrap logic from dpos/delegate_info to afterBlockApply step
        *   During the bootstrap period, it should not updateValidator
        *   Only at the end of the bootstrapPeriod, which should be end of the round, should call **updateValidator**
        *   It should still snapshot the voteWeight at the end of the round
    *   Move update logic from dpos/delegate_info to afterBlockApply step
        *   Standby delegate should have canVote false, else true
        *   Instead of calling setForgerList, it should call **updateValidator**

#### General concept

- afterGenesisBlockApply
   * Validate dpos related properties in genesis account
- afterBlockApply
   * Update delegate productivity
   * Snapshot voteWeight at the end of the round
   * Skip updateValidators during bootstrap period except last block of the period
   * Using the snapshot voteWeight, create validators list and call updateValidators
   * Cleanup finalized state data

### Motivation

DPoS module is a module to handle DPoS selection protocol

### Acceptance criteria

*   DPoS module should not be dependent on lisk-dpos (it will be removed)
*   All related unit tests are moved to module/dpos unit test

### Related Issues

#5603

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?

This PR resolves #5608

### How was it solved?

Added the transaction asset for vote transaction.

### How was it tested?

Run all unit tests.

### CODE REVIEW

File: framework/test/unit/specs/modules/dpos/transaction_assets/vote_transaction_asset.spec.ts [hunk 1]
```diff
@@ -0,0 +1,812 @@
...
+				await transactionAsset.applyAsset(applyInput);
+
+				const updatedDelegate1 = await stateStoreMock.account.get<DPOSAccountProps>(
+					delegate1.address,
+				);
+
+				expect(updatedDelegate1.dpos.delegate.totalVotesReceived).toEqual(
+					delegate1VoteAmount + positiveVoteDelegate1,
+				);
+			});
```
Comment (shuse2) [inline framework/test/unit/specs/modules/dpos/transaction_assets/vote_transaction_asset.spec.ts:522]: order of votes and unlocking should be checked?

Comment (nazarhussain) [reply framework/test/unit/specs/modules/dpos/transaction_assets/vote_transaction_asset.spec.ts:522]: For checking the order there are separate test cases both for `sentVotes` and `unlocking`

File: framework/src/modules/dpos/transaction_assets/vote_transaction_asset.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+				const index = originalUpvoteIndex > -1 ? originalUpvoteIndex : sender.dpos.sentVotes.length;
+				const upvote =
+					originalUpvoteIndex > -1
+						? sender.dpos.sentVotes[originalUpvoteIndex]
+						: {
+								delegateAddress: vote.delegateAddress,
+								amount: BigInt(0),
+						  };
+				upvote.amount += vote.amount;
+				// Special case for postgres because maximum is int64 for bigint in postgres
```
Comment (ishantiw) [inline framework/src/modules/dpos/transaction_assets/vote_transaction_asset.ts:189]: I think we can remove this comment now

Comment (ishantiw) [inline framework/src/modules/dpos/transaction_assets/vote_transaction_asset.ts:189]:
```suggestion
```

**Pull request merged**